app/models/manageiq/providers/inventory/persister/builder/persister_helper.rb

```diff
@@ -1,39 +1,8 @@
 module ManageIQ::Providers::Inventory::Persister::Builder::PersisterHelper
   extend ActiveSupport::Concern
 
-  # Interface for creating InventoryCollection under @collections
-  #
-  # @param builder_class    [ManageIQ::Providers::Inventory::Persister::Builder] or subclasses
-  # @param collection_name  [Symbol || Array] used as InventoryCollection:association
-  # @param extra_properties [Hash]   props from InventoryCollection.initialize list
-  #         - adds/overwrites properties added by builder
-  #
-  # @param settings [Hash] builder settings
-  #         - @see ManageIQ::Providers::Inventory::Persister::Builder.default_options
-  #         - @see make_builder_settings()
-  #
-  # @example
-  #   add_collection(ManageIQ::Providers::Inventory::Persister::Builder::CloudManager, :vms) do |builder|
-  #     builder.add_properties(
-  #       :strategy => :local_db_cache_all,
-  #     )
-  #   )
-  #
-  # @see documentation https://github.com/ManageIQ/guides/tree/master/providers/persister/inventory_collections.md
-  #
   def add_collection(builder_class, collection_name, extra_properties = {}, settings = {}, &block)
-    builder = builder_class.prepare_data(collection_name,
-                                         self.class,
-                                         make_builder_settings(settings),
-                                         &block)
-
-    builder.add_properties(extra_properties) if extra_properties.present?
-
-    builder.add_properties({:manager_uuids => references(collection_name)}, :if_missing) if targeted?
-
-    builder.evaluate_lambdas!(self)
-
-    collections[collection_name] = builder.to_inventory_collection
+    super(collection_name, builder_class, extra_properties, settings, &block)
```

Comment:
There are switched `builder_class` and `collection_name`. There is default value for `builder_class` in gem, which allows to define collection with name only, if we don't need (or don't have) shared IC definition in Builder